lib/library.rb

```diff
@@ -0,0 +1,51 @@
+class Library
+  attr_reader :name,
+              :books,
+              :authors,
+              :checked_out_books,
+              :checkout_tally
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @checkout_tally = Hash.new(0)
+  end
+
+  def add_author(author)
+    @authors.push(author)
+
+    author.books.each do |book|
+      @books.push(book)
+    end
```

Comment:
this is okay, but if an author writes a new book later, how do we add that book to our library?